addons/binding/org.openhab.binding.tadoac/src/main/java/org/openhab/binding/tadoac/handler/TadoACHandler.java

```diff
@@ -197,20 +198,21 @@ private synchronized void refreshSetting() {
                 }
                 updateState(CHANNEL_MODE, new DecimalType(mode));
             }
-            if (newSetting.getTemperature() != null) {
-                setting.setTemperature(newSetting.getTemperature());
-                updateState(CHANNEL_TEMP, new DecimalType(newSetting.getTemperature().getCelsius()));
+            TadoACTemperature newTemp = newSetting.getTemperature();
+            if (newTemp != null) {
+                setting.setTemperature(newTemp);
+                updateState(CHANNEL_TEMP, new DecimalType(newTemp.getCelsius()));
             }
+            updateStatus(ThingStatus.ONLINE);
         } catch (Exception e) {
-            logger.error("error while polling for new state", e);
+            logger.info("error while polling for new state", e);
+            updateStatus(ThingStatus.OFFLINE);
         }
     }
 
     @Override
     public void initialize() {
         setting = new TadoACSetting();
-        refreshFuture = scheduler.scheduleAtFixedRate(() -> refreshSetting(), 15, 60, TimeUnit.SECONDS);
-        updateStatus(ThingStatus.ONLINE);
-
+        scheduler.scheduleAtFixedRate(() -> refreshSetting(), 15, 60, TimeUnit.SECONDS);
```

Comment:
What about my recommendation to have the user decide which time interval to use for updating the state of the ac?